### PR TITLE
tests: add OTLP receiver/sink

### PR DESCRIPTION
**Pull request by rmfitzpatrick**
These changes introduce an `OTLPReceiverSink` helper type to the tests module capable of asserting against metrics received from a Collector process.

### CODE REVIEW

**Comment by tigrannajaryan** [comment]
It seems to overlap with OTLPDataReceiver in the testbed: https://github.com/open-telemetry/opentelemetry-collector/blob/90ccbfa94d5c33ed13ead47d8af04f357c81e999/testbed/testbed/receivers.go#L261

Is this sufficiently different to warrant the existence?

**Comment by tigrannajaryan** [comment]
I guess OTLPDataReceiver is more specialized for the testbed so maybe not. Just take a look at OTLPDataReceiver to make sure you know it exists.

**Comment by rmfitzpatrick** [comment]
> I guess OTLPDataReceiver is more specialized for the testbed so maybe not. Just take a look at OTLPDataReceiver to make sure you know it exists.

Thanks, @tigrannajaryan.  I wasn't familiar with it specifically, but am loosely familiar with the testbed usage in the metric correctness tests.  I don't see an immediately worthwhile way to include that in the combination sink, but they both stand up an otlp receiver in a similar way.

**Review by tigrannajaryan — APPROVED** [review]

File: tests/testutils/otlp_receiver_sink.go [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+	"go.opentelemetry.io/collector/component"
+	"go.opentelemetry.io/collector/component/componenttest"
+	"go.opentelemetry.io/collector/config/confignet"
+	"go.opentelemetry.io/collector/consumer/consumertest"
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.opentelemetry.io/collector/receiver/otlpreceiver"
+	"go.uber.org/zap"
+)
+
+const defaultOTLPEndpoint = "localhost:4317"
```
[the remark this answers is not in the document]
Comment (rmfitzpatrick) [reply tests/testutils/otlp_receiver_sink.go]: Made required

File: tests/testutils/otlp_receiver_sink.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+// To be used as a builder whose Build() method provides the actual instance capable of starting the OTLP receiver
+// providing received metrics to test cases.
+type OTLPReceiverSink struct {
```
[the remark this answers is not in the document]
Comment (rmfitzpatrick) [reply tests/testutils/otlp_receiver_sink.go]: renamed to include metrics, as they are the only supported type atm.

**Pull request merged**